rust/operator-binary/src/spark_k8s_controller.rs

```diff
@@ -253,6 +253,8 @@ fn pod_template(
     Ok(Pod {
         metadata: ObjectMetaBuilder::new()
             .name(container_name)
+            .ownerreference_from_resource(spark_application, None, Some(false))
```

Comment:
Could you please elaborate on how the settings `block_owner_deletion` and `controller` were chosen?
Ideally as a comment would be great! I have to admit i don't really understand these settings.

Reply:
After reading a bit i would have expected `.ownerreference_from_resource(spark_application, None, None)` but honestly all is fine for me when it works as intended ;)

Reply:
This reference is not pointing to a controller (this seems to be implemented by spark internally and it is not clear where exactly that is done) but only provides a UID that can used to clean up resources cleanly (specifically driver pods and related config maps) when the spark application is deleted.

Reply:
I'll add that as a comment.

Reply:
thx!

Reply:
Done. And I agree - `None` is better as it is then clear we are delegating the internal management to k8s.

Reply:
thx again!